src/index.js

```diff
@@ -54,48 +47,65 @@ class ReactPlayerLoader extends React.Component {
   }
 
   /**
-   * Called just after the component has mounted on
-   * the DOM (right after `render()`).Will only be
-   * called in the web browser.
+   * Loads a new player based on the current props.
    */
-  componentDidMount() {
-    this.mounted_ = true;
-    // we have to make a copy of onSuccess and
-    // onFailure here, because we have to pass
-    // our own callbacks to player-loader
-    // to determine success/failure.
+  loadPlayer() {
+
+    // If there is any player currently loaded, dispose it before fetching a
+    // new one.
+    this.disposePlayer();
+
+    // We need to provide our own callbacks below, so we cache these
+    // user-provided callbacks for use later.
     const userSuccess = this.props.onSuccess;
     const userFailure = this.props.onFailure;
 
     const options = Object.assign({}, this.props, {
       refNode: this.refNode,
       refNodeInsert: 'append',
       onSuccess: ({ref, type}) => {
-        // ignore success when not mounted
-        // and dispose the player
-        if (!this.mounted_) {
-          disposePlayer(ref);
+
+        // If the component is not mounted when the callback fires, dispose
+        // the player and bail out.
+        if (!this.isMounted_) {
+          this.disposePlayer(ref);
           return;
         }
+
+        // Store a player reference on the component.
         this.player = ref;
 
+        // Null out the player reference when the player is disposed from
+        // outside the component.
+        ref.on('dispose', () => {
```

Comment:
should this just call `this.disposePlayer()`?

Reply:
No, because it should only happen for in-page players (this is now fixed) and, also, because we don't want to call `dispose` again. This is only relevant if some external code disposes the player.